HACKER/HACKER/Sources/ViewControllers/ShoppingRankingViewController/ShoppingRankingViewController.swift

```diff
@@ -0,0 +1,293 @@
+      $0.snp.makeConstraints { make in
+        make.top.equalTo(self.view.safeAreaLayoutGuide).offset(11)
+        make.trailing.equalToSuperview().offset(-24)
+        make.width.equalTo(35)
+        make.height.equalTo(34)
+      }
+    }
+  }
+  func layoutInfoView() {
+    self.view.add(infoView) {
+      $0.backgroundColor = .clear
+      $0.setRounded(radius: 15)
+      $0.snp.makeConstraints { make in
+        make.top.equalTo(self.infoButton.snp.bottom).offset(4)
+        make.trailing.equalTo(self.infoButton)
+        make.width.equalTo(255)
+        make.height.equalTo(72)
+      }
+    }
+  }
+  func layoutInfoLabel() {
+    self.infoView.add(infoLabel) {
+      $0.setupLabel(text: "랭킹과 머리카락은 00:00시 정각에\n업데이트 됩니다.", color: .clear, font: .titleBold(ofSize: 16))
+      $0.numberOfLines = 2
+      $0.snp.makeConstraints { make in
+        make.centerX.centerY.equalToSuperview()
+      }
+    }
+  }
+  func layoutTabBarCollectionView() {
+    self.view.add(self.tabbarCollectionView) {
+      $0.backgroundColor = .white
+      $0.snp.makeConstraints { make in
+        make.top.equalTo(self.pageLogoImage.snp.bottom)
+        make.leading.trailing.equalToSuperview()
+        make.width.equalTo(self.screenWidth)
+        make.height.equalTo(self.screenWidth * 54/375)
+      }
+    }
+  }
+  func layoutSectionLineImage() {
+    self.view.add(sectionLineImage) {
+      $0.image = UIImage(named: "sectionLine")
+      $0.contentMode = .scaleAspectFit
+      $0.snp.makeConstraints { make in
+        make.top.equalTo(self.tabbarCollectionView.snp.bottom).offset(-6)
+        make.centerX.equalToSuperview()
+        make.width.equalTo(375)
+        make.height.equalTo(15)
+      }
+    }
+  }
+  func layoutPageCollectionView() {
+    self.view.add(pageCollectionView) {
+      $0.backgroundColor = .hackerWhite
+      $0.isPagingEnabled = true
+      $0.showsHorizontalScrollIndicator = false
+      $0.contentInsetAdjustmentBehavior = .never
+      $0.snp.makeConstraints { make in
+        make.top.equalTo(self.sectionLineImage.snp.bottom)
+        make.leading.trailing.bottom.equalToSuperview()
+      }
+    }
+  }
+  func shoppingSelected() {
+    NotificationCenter.default.post(name: NSNotification.Name("shoppingSelect"), object: nil)
+    self.pageCollectionView.scrollToItem(at: NSIndexPath(item: 0, section: 0) as IndexPath, at: .left, animated: true)
+  }
+  func rankingSelected() {
+    NotificationCenter.default.post(name: NSNotification.Name("rankingSelect"), object: nil)
+    self.pageCollectionView.scrollToItem(at: NSIndexPath(item: 0, section: 1) as IndexPath, at: .left, animated: true)
+  }
+  @objc func infoButtonClicked() {
+    if buttonNum == 0 {
+      self.infoView.backgroundColor = .hackerBlack
+      self.infoLabel.textColor = .hackerWhite
+      self.view.bringSubviewToFront(infoView)
+      buttonNum = 1
+    } else {
+      self.infoView.backgroundColor = .clear
+      self.infoLabel.textColor = .clear
+      buttonNum = 0
+    }
```

Comment:
우선 색으로만 툴팁이 보이는지 안 보이는지를 정해주니까 툴팁이 눈으로는 안 보일 때도 실제로는 그 자리에 존재하고 있어서 뒤에 랭킹 버튼이 안 눌리더라구! 그래서 isHidden 프로퍼티를 활용해서 잠깐 숨겨주는 게 좋을 거 같아 (색은 뷰, 라벨 정의하는 부분에서 clear 말고 실제 색으로 넣어주고!)
```suggestion
    if buttonNum == 0 {
      self.infoView.isHidden = false
      self.view.bringSubviewToFront(infoView)
      buttonNum = 1
    } else {
      self.isHidden = true
      buttonNum = 0
    }
```
<br/>

그리고 변수 이름이 buttonNum이다 보니까 처음에는 이게 뭐 하는 앤지 딱 바로 감이 안 잡히더라구
이렇게 flag의 용도로 쓰이는 애들은 Bool로 정의해주면 훨씬 좋고, bool은 주로 is--- 이런 형태로 네이밍을 해줄 때가 많아
그래서 내가 새로 이름을 지어보면 isToolTipShown이라고 지을 수 있을 거 같고, 코드를 읽어봐도 `if isToolTipShown {}` 이 부분이 그냥 영어 문장처럼 '만약 툴팁이 보여지고 있다면' 이렇게 읽힐 수 있어서 더 직관적이지
그리고 이렇게 Bool로 해주면 좋은 이유 또 하나는 버튼이 눌리면 값을 지정해줄 필요 없이 not 연산자 하나만으로 true -> false로, false -> true로 바꿔줄 수 있다는 거야!
지금까지 말한 걸 그대로 코드로 옮겨보면 아래처럼 쓸 수 있어

```suggestion
    isToolTipShown = !isToolTipShown
    if isToolTipShown {
      self.infoView.isHidden = false
      self.view.bringSubviewToFront(infoView)
    } else {
      self.infoView.isHidden = true
    }
```